frontend/lib/src/components/widgets/DateInput/DateInput.tsx

```diff
@@ -125,6 +125,17 @@ function DateInput({
 
   const maxDate = useMemo(() => getMaxDate(element), [element])
 
+  const enableQuickSelect = useMemo(() => {
+    if (!element.isRange) {
+      return false
+    }
+
+    // Since quick select allows to select ranges up to the past 2 years,
+    // we should only enable it if the min date is older than 2 years ago.
+    const twoYearsAgo = moment().subtract(2, "years").toDate()
+    return minDate < twoYearsAgo
+  }, [element.isRange, minDate])
```

Comment:
An alternative here might be to manually configure the quick selection options based on the min value, but since this is a quite minor feature it might not be worth the added effort.